ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/SynchronizerPerformanceOptions.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.eth.sync;
+
+import tech.pegasys.pantheon.util.uint.UInt256;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import com.google.common.base.Suppliers;
+import picocli.CommandLine;
+
+/**
+ * A collection of flags/config values that are performance related.
+ *
+ * <p>These are segregated into a separate class so that they are not considered standard options
+ * that are used to configure Pantheon. Generally speaking you will want to leave these options
+ * alone unless you know that configuring them will improve your performace.
+ *
+ * <p>Each of these options starts with <code>--X</code>. This is a Java convention communicating
+ * that there is no API stability commitment to these options. While these flags may persist on a
+ * long term bases it should be noted they may be added or removed in future updates and no
+ * announcement needs to be made for their modifications. Use these at your own risk.
+ *
+ * <p>All of these options are also hidden so that they will not show up in the CLI help screen.
+ */
+public class SynchronizerPerformanceOptions {
+
+  private static final Supplier<SynchronizerPerformanceOptions> cachedInstance =
+      Suppliers.memoize(SynchronizerPerformanceOptions::new);
+
+  public static SynchronizerPerformanceOptions get() {
+    return cachedInstance.get();
```

Comment:
Is there any way we can avoid using a singleton here? It's probably ok in the command line stuff since it's one-pantheon-per-vm but it's so much simpler to just have a blanket avoid singletons rule.

Reply:
Not easily in the general case.  For the synchronizer I can add this to the SynchronizerConfiguration.  For RocksDB... there is no good configuration to hang it off.  And then how much about the details of how pantheon works should the command shell know?

The best answer I see you may not like:  DI such as Guice or Dagger.  An external module could tell the DI module that it provides a set of configuration mixins.  These are singleton within the scope of the injector, so the RocksDB module can provide it and use it and the command shell just passes the values.  But even without Guide and Dagger we could do this in a generic fashion but we quickly produces the services that Guice and Dagger are already providing.

Reply:
In terms of how much the command shell should know about how pantheon works, I think its one job is to setup the configuration for Pantheon.  So anything we expose on the command line should be also available as a config option when you use "embedded" Pantheon via `RunnerBuilder`.